.github/workflows/checks.yml

```diff
@@ -0,0 +1,50 @@
+name: Checks
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  test_and_lint:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Cache/Restore Mint packages
+        id: mint-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/mint
+          key: ${{ runner.os }}-mint-${{ hashFiles('**/Mintfile') }}
+          restore-keys: ${{ runner.os }}-mint-
+
+      - name: Install package manager "Mint"
+        run: |
+          brew upgrade mint || brew install mint || true
```

Comment:
`|| true` usage implies, that if there are any problems they wil go unnoticed, and we'll be unable to troubleshoot.

Perhaps we could check if the package is installed (`brew list | grep mint` should be enough), and depending on the status code (0 - it is installed, anything but - not installed; we base it on the grep output as well) we either install or upgrade? that way, if something happens (eg network outage on gh, etc) we immediately know that this is an intermittent issue.

Reply:
Good idea, changed it